### PR TITLE
README.md: Elaborate on purpose of repo, fix link to overlay

### PR DESCRIPTION
**Pull request by t-lo**
# Elaborate on repo purpose

The README, last updated 2017, stated the purpose of this repository wrongly. This commit fixes that to avoid confusion. Also, a stale link in the README which still pointed to CoreOS' `coreos-overlay` repo was corrected and now points to the respective flatcar-linux repo.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,13 +1,16 @@
-This repository contains the CoreOS branches of Linux. A new branch is created for each kernel version the distribution is rebased onto. The currently shipped version is available in the [overlay repository](https://github.com/coreos/coreos-overlay/tree/master/sys-kernel/coreos-sources/files).
+This repository holds a fork of the upstream Linux kernel with Flatcar Linux specific changes on top. The repo is updated on an on-demand basis, i.e. when patches need to be ported to a newer kernel release, or when new patches need to be introduced.
```
Comment (vbatts) [inline README.md:1]: perhaps 
> Flatcar Container Linux (and historic CoreOS Container Linux) specific changes on top

File: README.md [hunk 2]
```diff
@@ -1,13 +1,16 @@
...
+Please note that this repo solely exists for development of Flatcar Linux specific patches for the [overlay repository](https://github.com/flatcar-linux/coreos-overlay/tree/master/sys-kernel/coreos-sources/files). This repository is not used by the build process at all; it merely exists for generating patchsets that live in coreos-overlay. The build process pulls the sources directly from `git.kernel.org`.
+
+
+To add a patch, check out the branch you want to introduce a change to (probably `v${VERSION}-flatcar`. or `v${VERSION}-coreos` for legacy branches), apply your change, and then run `git format-patch ${TAG}` where tag is the upstream kernel tag (i.e. without `-flatcar`). This generates a set of patches that can be used by `./revbump.sh` script in the `coreos-sources` directory in the overlay.
```
Comment (vbatts) [inline README.md:6]: > where `${TAG}` is the exact upstream kernel git tag

File: README.md [hunk 3]
```diff
@@ -1,13 +1,16 @@
...
+This repository holds a fork of the upstream Linux kernel with Flatcar Linux specific changes on top. The repo is updated on an on-demand basis, i.e. when patches need to be ported to a newer kernel release, or when new patches need to be introduced.
 
-This repository is not used by the build process at all; it merely exists for generating patchsets that live in coreos-overlay. To add a patch, check out the branch you want to add a patch to (probably v${VERSION}-coreos), apply the patch, then run `git format-patch ${TAG}` where tag is the upstream kernel tag (no "-coreos"). This generates a set of patches that can be used by `./revbump.sh` script in the `coreos-sources` directory in the overlay.
+Please note that this repo solely exists for development of Flatcar Linux specific patches for the [overlay repository](https://github.com/flatcar-linux/coreos-overlay/tree/master/sys-kernel/coreos-sources/files). This repository is not used by the build process at all; it merely exists for generating patchsets that live in coreos-overlay. The build process pulls the sources directly from `git.kernel.org`.
```
Comment (sayanchowdhury) [inline README.md:3]: s/Flatcar Linux/Flatcar Container Linux

**Comment by t-lo** [comment]
Thank you for your feedback! I've updated the PR, addressing the issues raised.

File: README.md (hunk 2 above)
Comment (vbatts) [reply README.md]: missed this

File: README.md (hunk 3 above)
Comment (vbatts) [reply README.md]: missed this

**Comment by vbatts** [comment]
:+1:

**Pull request merged**